openedx_learning/apps/core/itemstore/models.py

```diff
@@ -0,0 +1,133 @@
+covers basic metadata that is intrinsic to the item itself, and now how it's
+used in a LearningContext. ItemRaw represents the raw byte data.
+
+TODO: Add link to ADR after it merges.
+"""
+from django.db import models
+
+from openedx_learning.lib.fields import (
+    hash_field,
+    identifier_field,
+    immutable_uuid_field,
+    manual_date_time_field,
+)
+from ..publish.models import LearningContext, LearningContextVersion
+
+
+class Item(models.Model):
+    """
+    This represents any content that has ever existed in a LearningContext.
+    """
+    uuid = immutable_uuid_field()
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
+    identifier = identifier_field()
+
+    # TODO: These pointers to the latest published version are convenient, but
+    # we don't currently have the data integrity guarantees to make sure that 
+    # multiple versions aren't active at the same time. Maybe add a ref to the
+    # Item model in LearningContextVersionItemVersion, so we can force that
+    # constraint?
+    published_version = models.ForeignKey('ItemVersion', on_delete=models.RESTRICT, null=True, related_name="+")
+    first_published_at = models.DateTimeField(null=True)  # can be null if never published 
+    last_published_at = models.DateTimeField(null=True)   # can be null if never published
+
+    created = manual_date_time_field()
+    modified = manual_date_time_field()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["learning_context_id", "identifier"],
+                name="learning_publishing_lcb_one_identifier_per_lc",
+            )
+        ]
+
+    def __str__(self):
+        return f"{self.identifier}"
+
+
+class ItemVersion(models.Model):
+    """
+    A particular version of an Item.
+
+    A new ItemVersion is created anytime there is either a change to the content
+    or a change to the policy around a piece of content (e.g. schedule change).
+    """
+    uuid = immutable_uuid_field()
+    item_info = models.ForeignKey('ItemInfo', on_delete=models.RESTRICT)
+    item = models.ForeignKey(Item, on_delete=models.CASCADE)
+    title = models.CharField(max_length=1000, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.uuid}: {self.title}"
+
+
+class LearningContextVersionItemVersion(models.Model):
+    learning_context_version = models.ForeignKey(LearningContextVersion, on_delete=models.CASCADE)
+    item_version = models.ForeignKey(ItemVersion, on_delete=models.RESTRICT)
+
+
+class ItemInfo(models.Model):
+    """
+    Basic metadata about an Item.
+
+    This is mostly here to give supplmentary models to hang off of this and give
+    more interesting models based on the data in ItemRaw.
+    """
+    uuid = immutable_uuid_field()
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
+    type = models.CharField(max_length=100)
```

Comment:
Can we always deduce `type` from `item_raw.mime_type`?

Reply:
I had originally meant it as something different, but I'm not sure if that holds up in the latest iteration.

The difference I had in mind originally was something like an HTMLBlock's HTML. That it would have an `ItemInfo` that showed it was an XBlock with HTML handler, but the raw data had a MIME type of "text/html".